src/main/java/io/streamnative/pulsar/manager/mapper/TopicsStatsMapper.java

```diff
@@ -57,17 +69,17 @@
 
     @Select({"<script>",
             "SELECT environment, cluster, tenant, namespace, persistent, topic,"
-                + "sum(producerCount) as producerCount,"
-                + "sum(subscriptionCount) as subscriptionCount,"
-                + "sum(msgRateIn) as msgRateIn,"
-                + "sum(msgThroughputIn) as msgThroughputIn,"
-                + "sum(msgRateOut) as msgRateOut,"
-                + "sum(msgThroughputOut) as msgThroughputOut,"
-                + "avg(averageMsgSize) as averageMsgSize,"
-                + "sum(storageSize) as storageSize, timestamp FROM topicsStats",
-            "WHERE environment=#{environment} and tenant=#{tenant} and namespace=#{namespace} and timestamp=#{timestamp} and " +
+                + "sum(producer_count) as producerCount,"
+                + "sum(subscription_count) as subscriptionCount,"
+                + "sum(msg_rate_in) as msgRateIn,"
+                + "sum(msg_throughput_in) as msgThroughputIn,"
+                + "sum(msg_rate_out) as msgRateOut,"
+                + "sum(msg_throughput_out) as msgThroughputOut,"
+                + "avg(average_msg_size) as averageMsgSize,"
+                + "sum(storage_size) as storageSize, time_stamp as timestamp FROM topics_stats",
+            "WHERE environment=#{environment} and tenant=#{tenant} and namespace=#{namespace} and time_stamp=#{timestamp} and " +
                     "topic IN <foreach collection='topicList' item='topic' open='(' separator=',' close=')'> #{topic} </foreach>" +
-            "GROUP BY cluster, persistent, topic" +
+            "GROUP BY environment, cluster, tenant, namespace, persistent, topic, timestamp" +
```

Comment:
@sijie Please help review this SQL. I added several fields after group by.

Reply:
I had to make the same fix for HerdDB.
It is okay